participant/app/utils/routing.ts

```diff
@@ -11,13 +12,97 @@ export const routeRelative = (
   const url = new URL(request.url);
   const currentRouteParts: string[] =
     url.pathname.split("/") || ([] as string[]);
+  const currentLocation = currentRouteParts[currentRouteParts.length - 1];
+  const routeBase =
+    currentLocation === "recertify"
+      ? currentRouteParts
+      : currentRouteParts.slice(0, -1);
   if (params) {
     const queryString = stringify(params);
-    return `${[...currentRouteParts.slice(0, -1), target].join(
-      "/"
-    )}?${queryString}`;
+    return `${[...routeBase, target].join("/")}?${queryString}`;
   }
-  return [...currentRouteParts.slice(0, -1), target].join("/");
+  return [...routeBase, target].join("/");
+};
+
+export const throwRouteChangesRelative = (
+  request: Request,
+  currentLocation: string,
+  target: string,
+  acceptableRoutes?: string[]
+) => {
+  if (acceptableRoutes?.includes(currentLocation)) {
+    return true;
+  }
+  if (request.url.includes(target)) {
+    return true;
+  }
```

Comment:
To make sure I understand: when would the first `if` be true vs when the second `if` would be true?

Reply:
The first is "I'm on an acceptable page"

The second is "I'm where i was supposed to be sent to"

This code doesn't add where you are now to the list of acceptable pages until _after_ you complete the form and add the data to the sum of the submission Data

So, in effect, if you have no `name` data, you can be on the list of acceptableRoutes ("", "/recertify", "/about"), or you can be on the "/name" page so that you can actually fill out the name data

After you fall through that first conditional:
`if (!submissionData.name)`
We push "name" onto the list of acceptableRoutes

Then the acceptableRoutes is ("", "recertify", "about", "name")
If you don't have participant data, you can be on the "count" page or the "details" page, but this is the only state you can land on the "count" page with. As soon as you've saved participant data, you can no longer route to "count" -- so we never push that onto the acceptableRoutes list..

Reply:
Ah perfect. Thank you!